processing/src/main/java/org/apache/carbondata/processing/datatypes/ArrayDataType.java

```diff
@@ -322,4 +325,16 @@ public void getComplexColumnInfo(List<ComplexColumnInfo> columnInfoList) {
             name, false));
     children.getComplexColumnInfo(columnInfoList);
   }
+
+  @Override
+  public int getDepth() {
+    if (depth == 0) {
+      // calculate only one time
+      List<ComplexColumnInfo> complexColumnInfoList = new ArrayList<>();
```

Comment:
Why not call children.getDepth

Reply:
First time (For the first row (depth = 0), it calls `getComplexColumnInfo`, which recursively calls for child also.

I am reusing the existing function instead of writing one for depth.